### PR TITLE
BUILD-349: run builds CI for ocm, ocm-o and builder with tech preview enabled

### PR DESCRIPTION
**Pull request by jkhelil**
https://issues.redhat.com/browse/BUILD-349

### CODE REVIEW

**Comment by gabemontero** [comment]
run `make jobs` @jkhelil and commit/squash the changes from that 

they are picky about the ordering in the yaml files, hence the generated-config failure

*maybe* that is why none of the other jobs have reported status back yet

but you also need to make changes in the analogous locations in 
https://github.com/openshift/release/tree/master/ci-operator/jobs/openshift/builder
https://github.com/openshift/release/tree/master/ci-operator/jobs/openshift/openshift-controller-manager
https://github.com/openshift/release/tree/master/ci-operator/jobs/openshift/cluster-openshift-controller-manager-operator

where you define the details of your new e2e's

your def's there should be based off of the e2e-aws-builds e2e's ... as those are the extended tests we want to run

and you should also chagne the 4.10 files in addition to the master branch files 

after you make those changes, we'll see how the rehearsal jobs look, and go from there

**Comment by adambkaplan** [comment]
draft PRs don't launch CI jobs, convert this to a real PR.

**Comment by gabemontero** [comment]
> draft PRs don't launch CI jobs, convert this to a real PR.

yeesh didn't even know to look for that and overlooked that in the title

yeah need a real PR in addition the other items I mentioned

File: ci-operator/config/openshift/builder/openshift-builder-master.yaml
Comment (gabemontero) [inline ci-operator/config/openshift/builder/openshift-builder-master.yaml:80]: this is at least the correct workflow to start with from what I can tell @jkhelil 

but I suspect ultimately we will need to create a new workflow that merges openshift-e2e-aws-techpreview with 
openshift-e2e-aws-builds

we ultimately need a trigger of the build e2e's in openshift/origin

See https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/aws/builds/openshift-e2e-aws-builds-workflow.yaml#L12

**Review by gabemontero — CHANGES_REQUESTED** [review]
I think this is pretty close @jkhelil 

but I am pretty sure you need to create a new work profile that merges e2e-aws-builds and the tech preview one

see my comment below

**Comment by gabemontero** [comment]
you minimally need to fix the error noted in https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22451/pull-ci-openshift-release-master-step-registry-metadata/1445411228804452352 @jkhelil

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (gabemontero) [inline ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml:12]: not that I've done this before, but yes this is what I was suggesting @jkhelil 

now, over time, I suspect we'll create a new test suite in openshift/origin for build tests that we only run in the tech preview mode, but IMO at least that is outside the scope of your Jira.

when build csi volumes is completed, they'll need to make those updates and tweak this line

**Review by gabemontero — CHANGES_REQUESTED** [review]
Add your Jira identifier to the beginning of the PR title @jkhelil

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (jkhelil) [reply ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml:12]: alright, thanks @gabemontero

**Comment by gabemontero** [comment]
ok looking at some of the rehearsal failures, they are of the "known flake documented in sippy" variety

where all the sig-builds tests passed

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22451/rehearse-22451-pull-ci-openshift-cluster-openshift-controller-manager-operator-master-openshift-e2e-aws-builds-techpreview/1445624616859996160

for example .... given that, we do have the option of merging this even if there are rehearsals that are technically failing, if we are convince the failures are unrelated to our change here

but I see the same in all three repos, both master and 4.10

Now, confirming that the cluster is in fact a "tech preview" cluster ....

for that, I believe you need to open up must-gather @jkhelil and confirm.  I'll give you an example to get you started.

Here are the artifacts from one of your rehearsals:  https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/22451/rehearse-22451-pull-ci-openshift-openshift-controller-manager-release-4.10-e2e-aws-techpreview-featuregate/1445364991975755776/artifacts/e2e-aws-techpreview-featuregate/

I see some evidence there, but you should confirm for each of the master/4.10 builder/ocm/ocm-o permutations you have.

Also, if you look at the must-gather, you should be able to find the feature gate cluster resource.  I looked at one and things seem "OK", but you as PR author should take a stab at it for all 6 combinations.  Certainly reach out to me if you get stuck and need a pointer.

After that, I think we are at a point where @adambkaplan chimes in on approving PRs, see if he is happy with the names you picked for things, etc.

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (adambkaplan) [reply ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml:12]: > now, over time, I suspect we'll create a new test suite in openshift/origin for build tests that we only run in the tech preview mode, but IMO at least that is outside the scope of your Jira.

Alternatively, our tests can check if tech preview has been enabled, invoking appropriate skips if tech preview is or is not enabled.

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (adambkaplan) [inline ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml:6]: We also have a step that adds the github secrets needed for private git tests:
```suggestion
    - chain: ipi-aws-pre-techpreview
    - ref: build-github-secrets
    test:
```

**Comment by gabemontero** [comment]
> 
> @gabemontero i can see in must-gather.log ClusterOperators: clusteroperator/kube-apiserver is not upgradeable because FeatureGatesUpgradeable: "TechPreviewNoUpgrade" does not allow updates So clusters for these tests are techpreview clusters; 

Yes, you did find something that confirms that the tech feature gate is on.  And I'll also note since that message seems like it is citing "a prroblem", I am pretty sure that message is expected @jkhelil when using tech preview.

Just so you know, I confirmed it was a tech preview cluster by inspecting the `featuregates.yaml` file in the must-gather.tar file

In that tar file, the path is something like 

`/registry-build02-ci-openshift-org-ci-op-tt7ilggz-stable-sha256-864d8efffae65397e29d7bc8ee658a86d05ec8858133697cbace741a667439d6/cluster-scoped-resources/config.openshift.io/featuregates.yaml`

where you get at the tar file via this subpath from the artifacts link in your e2e test link ... something like 

`artifacts/openshift-e2e-aws-builds-techpreview/gather-must-gather/artifacts/must-gather.tar`

you see the tech preview feature gate set noted there

> I am bit annoyed with this test ci/rehearse/openshift/openshift-controller-manager/release-4.10/e2e-aws-techpreview-featuregate because it is not part of changes, it is linked ot an old commit before I fix and use openshift-e2e-aws-builds-techpreview rather than simple e2e-aws-techpreview-featuregate , so wondering why this test is still there
> 
> Do you know anyway to clean that. @gabemontero Also checked the failing tests and indeed these are flakes reported in sippy

I do not @jkhelil .... search the CI doc and/or ask the helpdesk over in #forum-testplatform if you really want to know how.  That said, as I mentioned before, we can merge this with some rehearsal failures if we are all agreed they are unrelated to your latest changes.

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (adambkaplan) [inline ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml:10]: Why do we need to gather the core dump? `ipi-aws-post` gives us the must-gather archive, which is sufficient for our testing needs.
```suggestion
    - chain: ipi-aws-post
```

**Review by adambkaplan — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (gabemontero) [reply ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml:12]: yep another valid option

File: ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml
Comment (jkhelil) [reply ci-operator/step-registry/openshift/e2e/aws/builds/techpreview/openshift-e2e-aws-builds-techpreview-workflow.yaml]: it is not needed indeed

**Comment by gabemontero** [comment]
> @adambkaplan @gabemontero I can see here that this alert is expected in early tests https://github.com/openshift/origin/blob/master/test/extended/prometheus/prometheus.go#L523 the comment in this line is explicit that this is expected // On a TechPreviewNoUpgrade cluster we must ignore the TechPreviewNoUpgrade alert // fired by the Kube API Operator. This alert is expected in this case. and the alert is in the allowedAlertNames, Perhaps we shouldnt do somethig about it can you confirm please ?

Yeah after revisiting everything @jkhelil my earlier post that listed `TechPreviewNoUpgrade` was not the relevant log

Bringing up a most recent failure, I see 

```


  | 2m32s
-- | --
fail [github.com/onsi/ginkgo@v4.7.0-origin.0+incompatible/internal/leafnodes/runner.go:113]: Oct  7 17:39:40.899: Unexpected alerts fired or pending after the test run:  alert ClusterNotUpgradeable fired for 1961 seconds with labels: {condition="Upgradeable", endpoint="metrics", name="version", severity="warning"}


```

So yes, confirmed, do what @adambkaplan specified.  Get a openshift/origin PR up that adds `ClusterNotUpgradeable` in the same fashion as `TechPreviewNoUpgrade` over at https://github.com/openshift/origin/blob/master/test/extended/prometheus/prometheus.go#L523

**Comment by jkhelil** [comment]
thank you @gabemontero 
Seems I missed a test on this so fixing this in this PR https://github.com/openshift/origin/pull/26515

**Pull request merged**
